frontend/src/lib/StaticGraphParser.test.ts

```diff
@@ -780,4 +787,56 @@ describe('StaticGraphParser', () => {
       ]);
     });
   });
+
+  function newRedundantWorkflow(): any {
+    return {
+      spec: {
+        entrypoint: 'redundant-pipeline',
+        templates: [
+          {
+            dag: {
+              tasks: [
+                { name: 'task-1', template: 'container-1' },
+                {
+                  dependencies: ['task-1'],
+                  name: 'task-2',
+                  template: 'container-2',
+                },
+                {
+                  dependencies: ['task-1', 'task-2'],
+                  name: 'task-3',
+                  template: 'container-3',
+                },
+              ],
+            },
+            name: 'redundant-pipeline',
+          },
+          { container: {}, name: 'container-1' },
+          { container: {}, name: 'container-2' },
+          { container: {}, name: 'container-3' },
+        ],
+      },
+    };
+  }
+
+  describe('transitiveReduction', () => {
+    it('reduces redundant workflow', () => {
+      const g = createGraph(newRedundantWorkflow());
+      const reduced = transitiveReduction(g);
+      expect(g.nodeCount()).toEqual(3);
+      expect(g.edgeCount()).toEqual(3);
+      expect(reduced).not.toBeUndefined();
```

Comment:
Simplified this to `toBeDefined()`. Reference: https://jestjs.io/docs/en/expect#tobedefined